sklearn/utils/validation.py

```diff
@@ -1187,6 +1207,13 @@ def check_X_y(
         Whether an array will be forced to be fortran or c-style. If
         `None`, then the input data's order is preserved when possible.
 
+    writeable : True or None, default=None
```

Comment:
I think accepting `True` but not `False` is a bit weird. Can we use more descriptive names? For example:

```suggestion
    writeable : "force" or "preserve", default="preserve"
```

Reply:
I agree that it's a bit weird. This is because I wanted to give it a similar status as `order` and `dtype`. I think a better option would have been `ensure_writeable = True/False` but the existing parameters with this naming pattern (ensure_xxx) are just to enable or disable a check, not to act on the output array.

Maybe we could go with `force_writeable = True/False` or `make_writeable=True/False`. There's already `force_all_finite` that is just to enable a check but arguably it should be `ensure_all_finite`. What do you think ? I think my preference is `force_writeable` and rename `force_all_finite` into `ensure_all_finite` later.

Reply:
For this PR, I like `force_writeable`.

Can you open an issue on `ensure_all_finite` vs `force_all_finite`?

Reply:
Alright, I made the change and opened https://github.com/scikit-learn/scikit-learn/issues/29262